examples/step-93/doc/intro.dox

```diff
@@ -0,0 +1,369 @@
+make this connection. Further, that way would need to be be explicit (by way of a
+function that translates from DoF index to row and column, as already mentioned above) or there
+would be no end to the possibility of introducing bugs.
+
+In other words, the fact that we want to have $2n$ degrees
+of freedom in the DoFHandler, but matrices and vectors with
+$2n + m$ rows and columns does not work with deal.II's
+underlying design decisions. Creating matrices and vectors that happen
+to be larger is going to trigger lots of assertions, and for good
+reasons.
+
+
+
+<h3>How do we address this?</h3>
+
+So then how are we going to address this? At its core, what needs to
+happen is that the DoFHandler knows about the additional degrees of
+freedom. The question is how we are going to teach it about these
+"nonlocal" degrees of freedom -- they are, after all, not ones that
+are associated with vertices, edges, faces, or cells of the mesh as
+the usual finite element degrees of freedom are. The approach we are
+going to use is not what one would generally call "obvious" or
+"natural", but it does fit into the infrastructure deal.II provides
+(even though this infrastructure was originally built for other
+purposes). In fact, at the time this program is written in 2024, there
+is substantial debate on whether we can implement *better* ways to achieve
+the same result that do not require this kind of "abuse" of features intended
+for other purposes; if consensus is ever reached on these alternative
+approaches, this program may also be converted.
+
+Whatever the future, let us turn to how our goal can be achieved today.
+The key to the following is that deal.II (i) allows using different
+finite elements on different cells, originally intended to support
+$hp$ finite elements; and (ii) has elements that have no degrees of
+freedom. We have used these features in other tutorial programs:
+step-27 and step-75 make use of feature (i) to implement the $hp$
+finite element method, and step-46 makes use of feature (ii) for
+multiphysics problems where some
+variables live only on some cells.
+
+We will (ab)use the general approach used in step-46 for our purposes
+here. Specifically, on the first four cells of the mesh, we will
+define a finite element that has three components: a temperature component and a
+Lagrange multiplier component (both of which are the usual FE_Q
+element), plus a third vector component that is piecewise constant
+(FE_DGQ(0)). On all other
+cells, the finite element used has the same temperature and Lagrange
+multiplier components, but the third component is of type FE_Nothing
+and so has no degrees of freedom at all. Note that with this setup,
+the first four cells will have one additional degree of freedom.
+The end result of this
+approach is that nearly every function you have ever used will work as
+one would expect. In particular:
+
+- We have exactly four additional degrees of freedom, located on the
+  first four cells.
```

Comment:
Do we have any advice for how to make this work in parallel? One possibility would be to check for `global_active_cell_index() < 4`.